support-models/src/main/scala/com/gu/support/workers/PaymentFields.scala

```diff
@@ -10,7 +10,9 @@ sealed trait PaymentFields
 
 case class PayPalPaymentFields(baid: String) extends PaymentFields
 
-case class StripePaymentFields(stripeToken: String) extends PaymentFields
+sealed trait StripePaymentFields extends PaymentFields
+case class StripeSourcePaymentFields(stripeToken: String) extends StripePaymentFields // pre SCA compatibility
+case class StripePaymentMethodPaymentFields(paymentMethod: String) extends StripePaymentFields
```

Comment:
once we are not using the old mechanism any more, we can remove the stripeToken side of the type, and flatten it back down.